codeCore/src/prog8/code/target/encodings/KatakanaEncoding.kt

```diff
@@ -0,0 +1,94 @@
+package prog8.code.target.encodings
+
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Ok
+import com.github.michaelbull.result.Result
+import java.io.CharConversionException
+import java.nio.charset.Charset
+
+object KatakanaEncoding {
+    val charset: Charset = Charset.forName("JIS_X0201")
+
+    fun encode(str: String): Result<List<UByte>, CharConversionException> {
+        return try {
+            val mapped = str.map { chr ->
+                when (chr) {
+                    // TODO: Convert regular katakana to halfwidth katakana (java lib doesn't do that for us 
+                    //       and simply returns '?' upon reaching a regular katakana character)
+                    //       NOTE: we probably need to somehow do that before we reach this `when`, 
+                    //             as one regular katakana character often results in two HW katakana characters
+                    //             due to differences in how diacritics are handled.
+
+                    '\u0000' -> 0u
+                    '\u00a0' -> 0xa0u // $a0 isn't technically a part of JIS X 0201 spec, and so we need to handle this ourselves
+                    
+                    // Tsu can be interpreted as a smiley
+                    '🙂' -> 0xc2u
```

Comment:
This is not valid Kotlin syntax unfortunately.
The error caused is: " Too many characters in a character literal ''🙂'' "

The smileys and the running man emoji appear to require multiple characters in their representation (a unicode surrogate pair).
I am afraid we need to get rid of them in this literal translation table...

Reply:
yeah, whatever, your code, do whatever is required